File: 11-Container-With-Most-Water/maxArea.js
var maxArea = function(height) {
    let left = 0;
    let right = height.length - 1;
    let ans = 0;
    
    while (left < right) {
        let area = Math.min(height[left], height[right]) * (right - left);
        ans = Math.max(ans, area);
        
        if (height[left] > height[right]) {
            right--;
        }
        else {
            left++;
        }
    }
    
    return ans;
};

File: 11-Container-With-Most-Water/maxArea.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        ans = 0
        
        while left < right:
            area = min(height[left], height[right]) * (right - left)
            ans = max(ans, area)
            
            # Since area is constrained by the shorter side, keep the longer side
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
            
        return ans